### PR TITLE
Feature/tca 549/jump links additional funcs 1

### PR DESCRIPTION
**Pull request by ampaveliev**
**Related**
https://oat-sa.atlassian.net/browse/TCA-549

**Description**
Jumpl links changes. For more information please have a look at the [Jira ticket](https://oat-sa.atlassian.net/browse/TCA-549) _Acceptance Criteria_ section

**How to test**
1. Create test with several section. Some of them should be without review panel
2. Start test as test taker
3. Check Jump links
4. Switch between sections and check status JL
5. Switch between items, answer for some of them and check Question JL
6. Try to flag / unflag item and check question JL again

### CODE REVIEW

File: src/plugins/content/accessibility/jumplinks/helpers.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return broker
+            .getPanelArea()
+            .find(':not(.hidden)[tabindex]')
+            .first();
+    },
+});
+
+export const getItemStatus = (item) => {
+    if (item.flagged) {
+        return 'Flagged for review';
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/helpers.js:48]: This should be translated
`__('Flagged for review')`

File: src/plugins/content/accessibility/jumplinks/helpers.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    if (item.answered) {
+        return 'Answered';
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/helpers.js:52]: `__('Answered')`

File: src/plugins/content/accessibility/jumplinks/helpers.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    if (item.viewed) {
+        return 'Not answered';
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/helpers.js:56]: `__('Not answered')`

File: src/plugins/content/accessibility/jumplinks/helpers.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    return 'Not seen';
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/helpers.js:59]: `__('Not seen')`

File: src/plugins/content/accessibility/jumplinks/jumplinks.tpl [hunk 5]
```diff
@@ -1,15 +1,15 @@
 <nav class="jump-links-box" aria-label="{{__ "Jump Menu"}}">
     <ul>
         <li class="jump-link-item">
-            <button data-jump="question" class="jump-link" >{{__ "Jump to:"}} <b>{{__ "Question"}}</b></button>
+            <button data-jump="question" class="jump-link" >{{__ "Jump to:"}} <b>{{__ "Question"}} - {{__ questionStatus}}</b></button>
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/jumplinks.tpl:4]: Translation requires pure string. Using a variable or any other dynamic input will prevent the scanner to fetch the text. For such situation, text should be translated upfront.

File: src/plugins/content/accessibility/jumplinks/plugin.js [hunk 6]
```diff
@@ -50,40 +53,81 @@ export default pluginFactory({
...
+                    .parent();
+                if (enabled) {
+                    panelJumplink.removeClass('hidden');
+                } else {
+                    panelJumplink.addClass('hidden');
+                }
+            })
+            .on('changeQuesitionStatus', function changeQuesitionStatus(questionStatus) {
+                const elem = this.getElement();
+                const text = __('Question') + ' - ' + __(questionStatus);
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/plugin.js:99]: Translation has to be done upfront. We must not use variable or any other dynamic input for the `__()` helper.
Moreover, you should rather use string literals here to compose the string. Strange that your IDE didn't catch that.

File: src/plugins/content/accessibility/jumplinks/plugin.js [hunk 7]
```diff
@@ -50,40 +53,81 @@ export default pluginFactory({
...
+            })
+        this.shortcuts = shortcutsFactory({});
     },
 
     /**
      * Called during the runner's render phase
      */
     render: function render() {
-        this.jumplinks.render(this.getAreaBroker().getControlArea());
+        const jumplinksContainer = $('<div class="jump-links-container"></div>');
```
Comment (jsconan) [inline src/plugins/content/accessibility/jumplinks/plugin.js:128]: Inline template must be avoided. Please use a proper Handlebars template for any HTML to inject.

File: src/plugins/content/accessibility/keyNavigation/strategies/jumpLinks.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    name: 'jump-links',
+
+    /**
+     * Builds the top toolbar navigation strategy.
+     *
+     * @returns {keyNavigationStrategy}
+     */
+    init() {
+        const config = this.getConfig();
+        const $jumpLinksBox = $('.jump-links-container');
```
Comment (jsconan) [inline src/plugins/content/accessibility/keyNavigation/strategies/jumpLinks.js:47]: Cannot you rely on a scoped container instead? This code is not resilient against multiple instances of the same component in the same page.

File: src/plugins/content/accessibility/jumplinks/helpers.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2020 (original work) Open Assessment Technologies SA ;
+ */
+
+export const getJumpElementFactory = (broker) => ({
```
[the remark this answers is not in the document]
Comment (ampaveliev) [reply src/plugins/content/accessibility/jumplinks/helpers.js:19]: You are right, thx

File: src/plugins/content/accessibility/jumplinks/helpers.js (hunk 1 above)
Comment (ampaveliev) [reply src/plugins/content/accessibility/jumplinks/helpers.js:48]: Ok, I'll add it here

File: src/plugins/content/accessibility/jumplinks/jumplinks.tpl (hunk 5 above)
Comment (ampaveliev) [reply src/plugins/content/accessibility/jumplinks/jumplinks.tpl:4]: Never thought about it. Thx

**Pull request merged**